packages/platform/src/Url.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @since 1.0.0
+ */
+import * as Either from "effect/Either"
+import { dual } from "effect/Function"
+import * as Error from "./Error.js"
+import * as UrlParams from "./UrlParams.js"
+
+/** @internal */
+const immutableSetter = <M>(clone: (mutable: M) => M) =>
+<P extends keyof M>(property: P): {
+  (value: M[P]): (mutable: M) => M
+  (mutable: M, value: M[P]): M
+} =>
+  dual(2, (mutable: M, value: M[P]) => {
+    const result = clone(mutable)
+    result[property] = value
+    return result
+  })
+/** @internal */
```

Comment:
```suggestion
  })

/** @internal */
```

Some whitespace between apis would be good.